components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -566,35 +566,49 @@ fn animate_discrete<T: Clone>(this: &T, other: &T, procedure: Procedure) -> Resu
 
 impl Animate for AnimationValue {
     fn animate(&self, other: &Self, procedure: Procedure) -> Result<Self, ()> {
-        let value = match (self, other) {
-            % for prop in data.longhands:
-            % if prop.animatable:
-            % if prop.animation_value_type != "discrete":
-            (
-                &AnimationValue::${prop.camel_case}(ref this),
-                &AnimationValue::${prop.camel_case}(ref other),
-            ) => {
-                AnimationValue::${prop.camel_case}(
-                    this.animate(other, procedure)?,
-                )
-            },
-            % else:
-            (
-                &AnimationValue::${prop.camel_case}(ref this),
-                &AnimationValue::${prop.camel_case}(ref other),
-            ) => {
-                AnimationValue::${prop.camel_case}(
-                    animate_discrete(this, other, procedure)?
-                )
-            },
-            % endif
-            % endif
-            % endfor
-            _ => {
+        use self::AnimationValue::*;
+
+        Ok(unsafe {
+            #[repr(C)]
+            struct AnimationValueVariantRepr<T> {
+                tag: u16,
+                value: T
+            }
+
+            let this_tag = *(self as *const _ as *const u16);
```

Comment:
All this `tag` stuff could probably just call `id()`, wdyt?

Reply:
Yeah but I just kept thing consistent with `PropertyDeclaration` and it makes the invariant needed for this code to be sound more explicit (that the discriminants must correspond in the representation).